service/src/test/java/com/kodemon/service/test/facade/FightFacadeTest.java

```diff
@@ -0,0 +1,301 @@
+        List<FightDTO> fights = fightFacade.listFightsBetween(from, to);
+        assertThat(fights.size(), is(2));
+    }
+
+    @Test
+    public void listTodaysFightTest() {
+        Date today = new Calendar.Builder().setDate(2015, 4, 1).build().getTime();
+        DateTime currentDate = new DateTime(today);
+        Date dayStart = currentDate.withTimeAtStartOfDay().toDate();
+        Date dayEnd = currentDate.plusDays(1).withTimeAtStartOfDay().toDate();
+        when(timeService.currentDate()).thenReturn(today);
+        when(timeService.startOfTheDay(today)).thenReturn(dayStart);
+        when(timeService.endOfTheDay(today)).thenReturn(dayEnd);
+        when(trainerFightService.findByFightTimeBetween(dayStart, dayEnd)).thenReturn(Collections.singletonList(fight1e));
+        when(beanMappingService.mapTo(Collections.singletonList(fight1e), FightDTO.class)).thenReturn(Collections.singletonList(fight1));
+
+        List<FightDTO> todaysFights = fightFacade.listTodaysFights();
+        assertThat(todaysFights.size(), is(1));
+    }
+
+    @Test
+    public void listAllFightsTest() {
+        when(trainerFightService.findAll()).thenReturn(trainerFights);
+        List<FightDTO> allFights = fightFacade.listAllFights();
+        assertThat(allFights.size(), is(2));
+    }
+
+    @Test
+    public void listFightsOfTrainerTest() {
+        when(trainerFightService.findByChallenger(challenger)).thenReturn(trainerFights);
+        List<FightDTO> fights = fightFacade.listAllFights();
+        assertThat(fights.size(), is(2));
+    }
+
+    @Test
+    public void listFightsOfGymTest() {
+        when(trainerFightService.findByTargetGym(targetGym)).thenReturn(trainerFights);
+        List<FightDTO> fights = fightFacade.listAllFights();
+        assertThat(fights.size(), is(2));
+    }
+
+    private void prepareTrainers() {
+        pikachu = new Pokemon();
+        pikachu.setName(PokemonName.PIKACHU);
+        pikachu.setLevel(20);
+        pikachu.setNickname("Yellow mouse");
+
+        pikachuDTO = new PokemonDTO();
+        pikachuDTO.setName(PokemonName.PIKACHU);
+        pikachuDTO.setLevel(20);
+        pikachuDTO.setNickname("Yellow mouse");
+
+        challenger = new Trainer();
+        challenger.setFirstName("Ash");
+        challenger.setLastName("Ketchum");
+        Date dob = new Calendar.Builder().setDate(1987, 4, 1).build().getTime();
+        challenger.setDateOfBirth(dob);
+        challenger.setUserName("Ash123");
+        challenger.addPokemon(pikachu);
+
+        challengerDTO = new UserDTO();
+        challengerDTO.setFirstName("Ash");
+        challengerDTO.setLastName("Ketchum");
+        challengerDTO.setDateOfBirth(dob);
+        challengerDTO.setUserName("Ash123");
+        challengerDTO.addPokemon(pikachuDTO);
+
+        pikachu.setTrainer(challenger);
+        pikachuDTO.setTrainer(challengerDTO);
+
+        when(beanMappingService.mapTo(pikachuDTO, Pokemon.class)).thenReturn(pikachu);
```

Comment:
Hm, I know this code (mocking/stubbing) does what it should, but unit tests should be independent and well, this is against this principle, moreover, you should specify the behavior for each test separately, because you want to have control over your code and know what is happening where. So please, omit mocking/stubbing here and move it to the test methods. And please reset the mocks in the `@AfterMethod` to ensure you have the behavior you want in each test.